src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\CustomUrl\Manager;
+
+use Ferrandini\Urlizer;
+use Sulu\Component\CustomUrl\Document\CustomUrlDocument;
+use Sulu\Component\CustomUrl\Repository\CustomUrlRepository;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\PathBuilder;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * Manages custom-url documents and their routes.
+ */
+class CustomUrlManager implements CustomUrlManagerInterface
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var CustomUrlRepository
+     */
+    private $customUrlRepository;
+
+    /**
+     * @var PathBuilder
+     */
+    private $pathBuilder;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        CustomUrlRepository $customUrlRepository,
+        PathBuilder $pathBuilder
+    ) {
+        $this->documentManager = $documentManager;
+        $this->customUrlRepository = $customUrlRepository;
+        $this->pathBuilder = $pathBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, array $data, $locale = null)
+    {
+        $document = $this->documentManager->create('custom_urls');
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist(
+            $document,
+            $locale,
+            [
+                'parent_path' => $this->getItemsPath($webspaceKey),
+                'node_name' => Urlizer::urlize($document->getTitle()),
```

Comment:
No, you just need to configure the path in the configuration for the
document alias, I forget how exactly, but is how the snippets already
work.

something like: $documentAlias => '/path/to/parent'


On Mon, Jan 18, 2016 at 05:14:34AM -0800, Wachter Johannes wrote:
>    In [1]src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php:
> 
>  > +    }
>  > +
>  > +    /**
>  > +     * {@inheritdoc}
>  > +     */
>  > +    public function create($webspaceKey, array $data, $locale = null)
>  > +    {
>  > +        $document = $this->documentManager->create('custom_urls');
>  > +        $this->bind($document, $data, $locale);
>  > +
>  > +        $this->documentManager->persist(
>  > +            $document,
>  > +            $locale,
>  > +            [
>  > +                'parent_path' => $this->getItemsPath($webspaceKey),
>  > +                'node_name' => Urlizer::urlize($document->getTitle()),
> 
>    but then i have to set the parent document. which would be an overhead i
>    think. if i use the parent_path here i dont have to load the parent.
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1927#discussion_r49996932
>    2. https://github.com/sulu-io/sulu/pull/1927/files#r49996932
>    3. https://github.com/sulu-io/sulu/pull/1927/files#r49996932